app/helpers/layout_helper.rb

```diff
@@ -347,6 +347,18 @@ def fullscreen_button(element = "$(this).prev()")
   private
 
   def authorized_associations(associations)
-    associations.included_modules.include?(Authorizable) ? associations.authorized : associations
+    if associations.included_modules.include?(Authorizable)
+      if associations.respond_to?(:klass)
+        associations.authorized(authorized_assocations_permission_name(associations.klass), associations.klass)
+      else
+        associations.authorized(authorized_assocations_permission_name(associations), associations)
+      end
+    else
+      associations
+    end
+  end
+
+  def authorized_assocations_permission_name(klass)
+    "view_#{klass.to_s.downcase.pluralize}"
```

Comment:
> Perhaps something for a unit or functional test, with an org that's associated to one of everything and a non-user, to show that the method shows every resource? It doesn't help as much with plugins, but ensures core is OK.

While I agree that such test would be nice to have, frankly I wouldn't get to it anytime soon, not speaking about the fact that this test should have been added prior this PR. I don't see a way how easily test "everything", whenever someone adds a new resource it should be added to the test.

> Seems a reasonable behaviour, OK. So Foo.authorized("unknown") would return resources for which the user has edit_foo for instance?

I was under that impression but when I tested it it seems I was wrong. `Foo.authorized("unknown")` returns `[]` but `Foo.authorized(nil)` or even `Foo.authorized(nil)` would return resources for which user has `edit_foo`. So what we could do is check whether a permission exists and return nil if not in `authorized_assocations_permission_name`. I suppose that would be better?